addon/mixins/run.js

```diff
@@ -77,7 +78,7 @@ export default Mixin.create({
     assert(`Called \`runTask\` on destroyed object: ${this}.`, !this.isDestroyed);
 
     let type = typeof callbackOrName;
-    let pendingTimers = this._getOrAllocateArray('_pendingTimers');
+    let pendingTimers = getOrAllocate(this, '_pendingTimers', []);
```

Comment:
I totally like this new more functional util style, but we have to be careful here.  This particular implementation creates a new array on _every_ invocation (regardless if it is needed).

We should probably just have two util methods (one is `getOrAllocateArray` and the other is `getOrAllocateObject` if we need it) to avoid the needless extra array creation...